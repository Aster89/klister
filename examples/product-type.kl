#lang kernel

(datatype (Ã— A B)
  (times A B))

(define fst
  (lambda (x)
    (case x
      [(times a _)
       a])))

(example (fst (times 'a 'b)))

(example fst)
