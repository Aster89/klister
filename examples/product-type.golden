#[product-type.kl:12.23-12.24]<a> : Syntax
#<closure> : ∀(α : *) (β : *). ((× α β) → α)
